src/main/java/com/iexec/common/docker/client/DockerClientInstance.java

```diff
@@ -278,7 +292,7 @@ public boolean pullImage(String imageName) {
             pullImageCmd
                     .withTag(repoAndTag.tag)
                     .exec(new PullImageResultCallback() {})
-                    .awaitCompletion(1, TimeUnit.MINUTES);
+                    .awaitCompletion(timeout.toSeconds(), TimeUnit.SECONDS);
             log.info("Pulled docker image [name:{}]", imageName);
             return true;
         } catch (Exception e) {
```

Comment:
From what I understand from the documentation:
```
    /**
     * Blocks until {@link ResultCallback#onComplete()} was called or the given timeout occurs
     * @return {@code true} if completed and {@code false} if the waiting time elapsed
     *         before {@link ResultCallback#onComplete()} was called.
     */
    public boolean awaitCompletion(long timeout, TimeUnit timeUnit) throws InterruptedException {
```
How about?
```
            boolean isPulledBeforeTimeout = pullImageCmd
                    [...]
                    .awaitCompletion(1, TimeUnit.MINUTES);
            if (!isPulledBeforeTimeout){
                log.error("Timeout [name:{}, timeout:{}]", ...);
            }
```

Reply:
Good idea. See https://github.com/iExecBlockchainComputing/iexec-common/pull/314/commits/22ef5e1c7a836cfc3c06c8f09f667b2df7f882b4.